ad7768-1fmcz/ad77681.c

```diff
@@ -243,10 +313,321 @@ int32_t ad77681_spi_read_adc_data(struct ad77681_dev *dev,
 
 	/* Fill the adc_data buffer */
 	memcpy(adc_data, buf, ARRAY_SIZE(buf));
+	
+	if (buf[1] & 0x80)
+		converted_data = (int32_t)(0xFF << 24 | (((buf[1]  &   0xFF)  << 16) |	
+												 ((buf[2]  &   0xFF)  << 8) |	
+												 ((buf[3]  &   0xFF)  << 0)));
+	else
+		converted_data = (int32_t)(0x00 << 24 | (((buf[1]  &   0xFF)  << 16) |	
+												 ((buf[2]  &   0xFF)  << 8) |	
+												 ((buf[3]  &   0xFF)  << 0)));
+	
+	Vref = ((double)(dev->vref)) / 1000.0;
+	*voltage = (double)(((2.0 * Vref) / AD7768_FULL_SCALE ) * converted_data);					// ((2*Vref)*code)/2^24	
+	
+	if ((*voltage > Vref) || (*voltage < Vref * (-1.0)))										// Overflow protection, voltage can only fit between +- Vref
+		ret = -1;
+
+	return ret;
+}
+
+/**
+ * This function uses continuous read mode, which has to be enabled in advance
+ * Read conversion result from device using GPIO DRDY interrupt in 16bit frames
+ *
+ *	    /-------------\       /-------------\       /-------------\
+ *------  24-bit data  -------  8-bit staus  -------   8-bit CRC   -------
```

Comment:
I would avoid creating diagrams from text characters for the header function comments. Here looks fine, but when Doxygen will parse the text, it will be messed up.